llvm/lib/CAS/OnDiskHashMappedTrie.cpp

```diff
@@ -660,7 +660,7 @@ OnDiskHashMappedTrie::recoverFromFileOffset(FileOffset Offset) const {
   // Check bounds.
   const uint64_t CurrentFileSize = Impl->File.getAlloc().size();
   if (Offset.get() > (int64_t)CurrentFileSize ||
-      Offset.get() + Impl->Trie.getRecordSize() >= CurrentFileSize)
+      Offset.get() + Impl->Trie.getRecordSize() > CurrentFileSize)
```

Comment:
Nitpick: `Impl->Trie.getRecordSize() > CurrentFileSize - Offset.get()` avoids a potential overflow, although if we overflow an `int64_t` it probably means something has gone horribly wrong already

Reply:
Good idea. Horrible things have happened before; might as well avoid the potential overflow :).

Reply:
Fixed in ce1d1a22853bcdffc49f127fe222c54ce827d49c.